app/dao/jobs_dao.py

```diff
@@ -125,58 +124,63 @@ def dao_update_job(job):
 
 
 def dao_get_jobs_older_than_data_retention(notification_types):
-    flexible_data_retention = ServiceDataRetention.query.filter(
-        ServiceDataRetention.notification_type.in_(notification_types)
-    ).all()
+    stmt = select(ServiceDataRetention).where(ServiceDataRetention.notification_type.in_(notification_types))
+    flexible_data_retention = db.session.scalars(stmt).all()
+
     jobs = []
     today = datetime.utcnow().date()
     for f in flexible_data_retention:
         end_date = today - timedelta(days=f.days_of_retention)
 
-        jobs.extend(Job.query.join(Template).filter(
+        stmt = select(Job).join(Template).where(
             func.coalesce(Job.scheduled_for, Job.created_at) < end_date,
-            Job.archived == False,  # noqa
+            Job.archived.is_(False),
```

Comment:
Better way?  Does `not Job.archived` work?

Reply:
`is_(False)` seems to be the correct way to do this.  As written, the WHERE clause is:
```
WHERE coalesce(jobs.scheduled_for, jobs.created_at) < :coalesce_1 AND jobs.archived IS false AND templates.template_type = :template_type_1 AND (jobs.service_id NOT IN (__[POSTCOMPILE_service_id_1])) ORDER BY jobs.created_at DESC
```
Replacing is_ with `not Job.archived` yields:
```
WHERE false ORDER BY jobs.created_at DESC
```